pypesto/objective/priors.py

```diff
@@ -0,0 +1,264 @@
+            return math.exp(x_log) * \
+                (d_log_f_dx(math.exp(x_log)) +
+                    math.exp(x_log) * dd_log_f_ddx(math.exp(x_log)))
+
+        return {'index': index,
+                'density_fun': log_f_log,
+                'density_dx': d_log_f_log,
+                'density_ddx': dd_log_f_log}
+
+    elif parameter_scale == 'log10':
+
+        log10 = math.log(10)
+
+        def log_f_log10(x_log10):
+            """log-prior for log10-parameters"""
+            return log_f(10**x_log10)
+
+        def d_log_f_log10(x_log10):
+            """derivative of log-prior w.r.t. log10-parameters"""
+            return d_log_f_dx(10**x_log10) * log10 * 10**x_log10
+
+        def dd_log_f_log10(x_log10):
+            """second derivative of log-prior w.r.t. log10-parameters"""
+            return log10**2 * 10**x_log10 * \
+                   (dd_log_f_ddx(10**x_log10) * 10**x_log10
+                    + d_log_f_dx(10**x_log10))
+
+        return {'index': index,
+                'density_fun': log_f_log10,
+                'density_dx': d_log_f_log10,
+                'density_ddx': dd_log_f_log10}
+
+    else:
+        raise ValueError(f"Priors in parameters in scale {parameter_scale}"
+                         f" are currently not supported.")
+
+
+def _prior_densities(prior_type: str,
+                     prior_parameters: np.array) -> [Callable, Callable, Callable]:
+    """
+    Returns a tuple of Callables of the (log-)density (in linear scale),
+    together with their first + second derivative (= senisis) w.r.t x
+    """
+
+    if prior_type == 'uniform':
+
+        log_f = _get_constant_function(-math.log(prior_parameters[1] - prior_parameters[0]))
+        d_log_f_dx = _get_constant_function(0)
+        dd_log_f_ddx = _get_constant_function(0)
+
+        return log_f, d_log_f_dx, dd_log_f_ddx
+
+    elif prior_type == 'normal':
+
+        sigma2 = prior_parameters[1]**2
+
+        def log_f(x):
+            return -math.log(2*math.pi*sigma2)/2 - \
+                   (x-prior_parameters[0])**2/(2*sigma2)
+
+        d_log_f_dx = _get_linear_function(-1/sigma2, prior_parameters[0]/sigma2)
+        dd_log_f_ddx = _get_constant_function(-1/sigma2)
+
+        return log_f, d_log_f_dx, dd_log_f_ddx
+
+    elif prior_type == 'laplace':
+        log_2_sigma = math.log(2*prior_parameters[1])
+
+        def log_f(x):
+            return -log_2_sigma - abs(x-prior_parameters[0])/prior_parameters[1]
+
+        def d_log_f_dx(x):
+            if x > prior_parameters[0]:
+                return -1/prior_parameters[1]
+            else:
+                return 1/prior_parameters[1]
+
+        dd_log_f_ddx = _get_constant_function(0)
+
+        return log_f, d_log_f_dx, dd_log_f_ddx
+
+    elif prior_type == 'logUniform':
+        # when implementing: add to tests
+        raise NotImplementedError
+    elif prior_type == 'logNormal':
+
+        sigma2 = prior_parameters[1]**2
```

Comment:
the mixture of sigma2 and prior_parameters[1] in these formulas is a bit confusing.